### ISSUE TITLE
Q: what is a hard-to-see 3D object?

### ISSUE THREAD

**Issue by benatuts** [RoboCupAtHome/RuleBook#567]
In "Carry my Luggage", what would be an example of a "hard to see 3D object"? Does this mean something made of transparent glass?

**Comment by moriarty** [comment]
Here is a list of properties:

1. Small (below the height of the lowest laser scanners in the league)
2. Transparent
3. Translucent
4. Psychedelic (an adversarial patch)
5. Retro-reflective
6. Reflective
7. Actively/Dyanamic

This is not a complete list. The object may have one of these properties, some of these properties, all of these properties or none of these properties.

We the TC, know what laser scanners and 3D cameras can and cannot see. Detection will be hard, and could require a combination of sensors.

Now, if I look around my apartment for examples of things which could be on the gound:

- A pacifier is small, and translucent.
- Childrens toys.
- A Roomba is short, and active/dynamic.
- A glass of water, or small espresso cup.
- A teaspoon.
- An Apple.
- A Safety vest, for cycling, is retro-reflective.
- A child’s play mat.
- My glasses.

The Roomba and Safety vest might be too large to be hard to see.

**Comment by benatuts** [comment]
Does hitting such objects constitute a "major collision", resulting in the robot being immediately stopped and "additional penalties may apply" (3.7.6)?

If it is intended that "detection will be hard", then shouldn't there be a "Deus ex Machina" rule for the hazard, especially given that it is a Stage I test?

Or.... can it be assumed that the operator will also avoid the obstacles (so that the operator is assumed to be following a safe trajectory)?

**Comment by moriarty** [comment]
In the past, I’ve seen a robot hit a small object which was below the laser scanner and the TC hit the estop and end the run.

No, IMHO there shouldn’t be an ex machina rule around them.

The operator will walk in a normal, reasonable manner.

**Comment by LoyVanBeek** [comment]
For example, in the past we've used stuff like these [bar]( https://www.yuyek.com/wp-content/uploads/2015/04/kitchen-marvellous-white-acrylic-circled-barstool-design-with-metal-base-and-footrest-24-modern-and-elegant-kitchen-bar-stools-to-inspire-you-ballard-design-bar-stool-height-bar-stools-contemporary-b.jpg) [stools](https://www.cultfurniture.com/images/products/zoom/1479389976-55931700.jpg). Among other things.

**Comment by benatuts** [comment]
Will they (the small object and the hard-to-see object) be known or unknown objects (in the sense of 3.3.5, or the training day furniture of 3.3.3)?

**Comment by johaq** [comment]
Hard to see objects will be part of the arena's furniture. Small objects can be from the object set or from the decorations in the arena. Which objects will not be announced before the test.

**Comment by kyordhel** [comment]
# TL;DR;
1. Small object can also be *unknown* objects.
2. Obstacles are bonus goals. If you're unsure the robot can handle it, opt out.

# Explanation
We expect robots to navigate **safely** indoors. Whatever idea you might have, **Security First** comes on top of it. Same as in *Clean Up*, the robot is assumed to detect and avoid running over objects on the floor (whatever they are), regardless whether it is going to pick them up as part of cleaning, or just evade them because is busy in another task (e.g. helping granny to reach the door).

I might have dropped my bifocal glasses last night and the last thing I want is a robot driving over them.

### Rules of thumb:
- If an object can get crushed, the robot will be stopped.
- If an object is being pushed forward a long way, the robot will be stopped.
- If an object is being pushed away, the robot may be stopped (based on referees criterion).

**Comment by johaq** [comment]
Announce it before the test. 
The objects will be chosen in a way that a team cannot specifically prepare for that object. Object avoidance should work for anything.
Maybe.
Currently no test allows a restart afaik. Maybe the section should be removed if we end up with no test allowing a restart.
If a team yells at me to stop the robot I will stop it. Test is over then though.
